src/OrchardCore.Modules/OrchardCore.Search.Lucene/Handler/LuceneIndexingContentHandler.cs

```diff
@@ -77,46 +77,67 @@ private static async Task IndexingAsync(ShellScope scope, IEnumerable<ContentCon
                 // Only process the last context.
                 var context = ContextsById.Last();
 
-                ContentItem published = null, latest = null;
-                bool publishedLoaded = false, latestLoaded = false;
+                ContentItem published = null;
 
+                if (context is PublishContentContext publishContext)
+                {
+                    if (publishContext.Cancel)
+                    {
+                        continue;
+                    }
+                    if (publishContext.PreviousItem != null)
+                    {
+                        published = publishContext.PreviousItem;
```

Comment:
I like the idea of using the Contexts to avoid to retrieve these later on. Though, I think that we are retrieving them back here in the Handler in the DB because historically we had issue as it could have been altered by another handler. Though, maybe not an issue anymore.